src/pages/product/product.tsx

```diff
@@ -0,0 +1,63 @@
+import { useContext, useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+import { IProduct } from 'interfaces/IProduct'
+import { UserContext } from 'contexts/userContext'
+import { findProduct, updateProduct } from 'contracts/products'
+import { ProductItems } from 'components/productItems'
+import styles from './styles.css'
+
+export const Product = () => {
+  const [product, setProduct] = useState<IProduct>({
+    id: 0,
+    title: '',
+    description: '',
+    image: '',
+    price: '0'
+  })
+  const { title } = useParams()
+  const { isAuthenticated } = useContext(UserContext)
+
+  const getData = async () => {
+    setProduct(await findProduct(title))
+  }
+
+  useEffect(() => {
+    getData()
+  }, [])
+
+  const handleSave = async (title: string, price: string, description: string) => {
+    if (title !== product.title || price !== product.price || description !== product.description) {
+      await updateProduct({
+        ...product,
+        title,
+        price,
+        description
+      })
+    }
+  }
+
+  return (
+    isAuthenticated
+      ? <ProductItems onSave={handleSave} title={product.title} price={String(product.price)} description={product.description === undefined ? 'No description' : product.description} />
```

Comment:
`description` should use the same logic for create and update. If there is no description show placeholder